src/librustdoc/html/static/js/search.js

```diff
@@ -669,42 +676,49 @@ function initSearch(rawSearchIndex) {
         skipWhitespace(parserState);
         let start = parserState.pos;
         let end;
-        if (parserState.userQuery[parserState.pos] === "[") {
+        if ("[(".indexOf(parserState.userQuery[parserState.pos]) !== -1) {
+            const endChar = parserState.userQuery[parserState.pos] === "[" ? "]" : ")";
+            const name = parserState.userQuery[parserState.pos] === "[" ? "[]" : "()";
+            const friendlyName = parserState.userQuery[parserState.pos] === "[" ? "slice" : "tuple";
```

Comment:
Instead of doing the same check three times, would it be better to instead write (although longer):

```suggestion
let endChar = ")";
let name = "()";
let friendlyName = "tuple";

if (parserState.userQuery[parserState.pos] === "[") {
    endChar = "]";
    name = "[]";
    friendlyName = "slice";
}
```

?